system/lib/fetch/asmfs.cpp

```diff
@@ -1910,7 +1910,7 @@ long __syscall220(int which, ...) // getdents64 (get directory entries 64-bit)
 	if (desc->file_pos <= file_pos)
 	{
 		de->d_ino = (ino_t)node; // TODO: Create inode numbers instead of using pointers
-		de->d_off = file_pos;
+		de->d_off = file_pos + sizeof(dirent);
```

Comment:
It looks like this code is for the first entry read , so shouldn't the d_off for that entry be 0?     There it looks like d_off is being set to the end of the entry?   Is this part of the change tested somewhere?

Reply:
This is tested implicitly in test_readdir.c by using telldir() but not by using d_off. d_off indeed points to the directory entry after. I can add that test as well!

Reply:
I think d_off is supposed to the point the beginning of the entry, not the end?  

I can see that this change would make sense when iterating backwards, but if iterate forwards than wouldn't we want d_off to point the start of the current record?  (In which case this change is not needed?)

Reply:
`d_off` should be the same value as `telldir()` after the call to `readdir()`, this isn't related to iterating forwards or backwards.

Reply:
See https://linux.die.net/man/2/getdents64

Reply:
Ah!  I see.